frontend/src/scenes/settings/environment/maxSettingsLogic.tsx

```diff
@@ -0,0 +1,90 @@
+import { actions, afterMount, kea, listeners, path, reducers } from 'kea'
+import { forms } from 'kea-forms'
+import { loaders } from 'kea-loaders'
+import api from 'lib/api'
+import { lemonToast } from 'lib/lemon-ui/LemonToast/LemonToast'
+
+import { CoreMemory } from '~/types'
+
+import type { maxSettingsLogicType } from './maxSettingsLogicType'
+
+export type CoreMemoryForm = {
+    text: string
+}
+
+export const maxSettingsLogic = kea<maxSettingsLogicType>([
+    path(['scenes', 'project', 'Settings', 'maxSettingsLogic']),
+
+    actions({
+        setIsLoading: (isLoading: boolean) => ({ isLoading }),
+    }),
+
+    reducers({
+        isLoading: [
+            false,
+            {
+                loadCoreMemory: () => true,
+                loadCoreMemorySuccess: () => false,
+                loadCoreMemoryFailure: () => false,
+            },
+        ],
+
+        isUpdating: [
+            false,
+            {
+                createCoreMemory: () => true,
+                createCoreMemorySuccess: () => false,
+                createCoreMemoryFailure: () => false,
+                updateCoreMemory: () => true,
+                updateCoreMemorySuccess: () => false,
+                updateCoreMemoryFailure: () => false,
+            },
+        ],
+    }),
+
+    loaders(({ values }) => ({
+        coreMemory: {
+            __default: null as CoreMemory | null,
+            loadCoreMemory: async () => {
+                const response = await api.coreMemory.list()
+                return response.results[0] || null
+            },
+            createCoreMemory: async (data: CoreMemoryForm) => {
+                const response = await api.coreMemory.create(data)
+                lemonToast.success('The Max’s memory has been created.')
+                return response
+            },
+            updateCoreMemory: async (data: CoreMemoryForm) => {
```

Comment:
This actually exposes a little too much of internal logic to the component. In the component, it shouldn't matter at all whether a `CoreMemory` row already exists in the database or not. In every case, `updateCoreMemory` should be the only concern, and the potential complexity of `createCoreMemory` should be folded into `updateCoreMemory`